src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class SalaryInfo {
+    private final Pattern datePattern = Pattern.compile("\\d{2}\\.\\d{2}\\.\\d{4}");
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
+        LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate toDate = LocalDate.parse(dateTo, formatter);
```

Comment:
Consider adding a try-catch block around the date parsing to handle potential `DateTimeParseException` errors gracefully. This will prevent the application from crashing if an invalid date format is provided.